src/ceph_fuse.cc

```diff
@@ -204,8 +204,15 @@ int main(int argc, const char **argv, const char *envp[]) {
           "client_try_dentry_invalidate");
 	bool can_invalidate_dentries =
           client_try_dentry_invalidate && ver < KERNEL_VERSION(3, 18, 0);
-	auto test_result = client->test_dentry_handling(can_invalidate_dentries);
-	int tr = test_result.first;
+        uint64_t max_retries = cct->_conf.get_val<uint64_t>(
+	  "client_max_retries_on_remount_failure");
+	std::pair<int, bool> test_result;
+	int i = 0, tr;
+	do {
+	  test_result = client->test_dentry_handling(can_invalidate_dentries);
+	  tr = test_result.first;
+	} while (i++ < max_retries && tr);
```

Comment:
do we need to wait before a retry ?

Reply:
Let's do the same with `Client.cc`, which will do this in the `tick` every second.

Reply:
Did this change actually make it? I do not find it convincing to do this on each tick. Is there a need for that?

Reply:
No need to do this per tick for this fix. When starting the ceph-fuse daemon it will create a thread to test the `remount` operation in parallel and it just needs to make sure that the `remount` work. As you know the `remount` operation will be used to invalidate the Dcache from kernel.

I added a wait before next retry just to make sure it won't be too fast and it will fail again and again if the kernel is not ready yet.